tools/ds-identify

```diff
@@ -1409,33 +1376,31 @@ dscheck_VMware() {
 
     # Check to see if running in a container and the VMware
     # datasource is configured via environment variables.
-    if vmware_has_envvar_vmx_guestinfo; then
-        if vmware_has_envvar_vmx_guestinfo_metadata || \
-            vmware_has_envvar_vmx_guestinfo_userdata || \
-            vmware_has_envvar_vmx_guestinfo_vendordata; then
+    if [ -n "${VMX_GUESTINFO}" ]; then
+        if [ -n "${VMX_GUESTINFO_METADATA}" ] || \
+           [ -n "${VMX_GUESTINFO_USERDATA}" ] || \
+           [ -n "${VMX_GUESTINFO_VENDORDATA}" ]; then
             return "${DS_FOUND}"
         fi
     fi
 
     # Do not proceed unless the detected platform is VMware.
-    if [ ! "${DI_VIRT}" = "vmware" ]; then
-        return "${DS_NOT_FOUND}"
-    fi
+    [ "${DI_VIRT}" = "vmware" ] || return ${DS_NOT_FOUND}
 
     # Do not proceed if the vmware-rpctool command is not present.
-    if ! vmware_has_rpctool; then
-        return "${DS_NOT_FOUND}"
-    fi
+    command -v vmware-rpctool >/dev/null 2>&1 || return ${DS_NOT_FOUND}
 
     # Activate the VMware datasource only if any of the fields used
     # by the datasource are present in the guestinfo table.
-    if { vmware_rpctool_guestinfo_metadata || \
-         vmware_rpctool_guestinfo_userdata || \
-         vmware_rpctool_guestinfo_vendordata; } >/dev/null 2>&1; then
-        return "${DS_FOUND}"
-    fi
+    local tmp=''
+    for tmp in "metadata" "userdata" "vendordata"; do
+      tmp="$(vmware-rpctool "info-get guestinfo.$tmp")"
```

Comment:
This is the tricky part. If we do `vmware-rpctool "info-set guestinfo.metadata ..."` once, even after doing `vmware-rpctool "info-set guestinfo.metadata  "` it won't go aways and `info-get guestinfo.userdata` returns 0 with empty string in return.

This remains same until poweroff. (reboot won't make it go).

If a user decides to use `datasource_list: ['VMware', 'NoCloud', 'OVF', 'ConfigDrive', 'OpenStack']`
VMware DS is always found but with no data and cloud-init stops searching for other DS because VMware is already found. So, we need to check for alpha numeric string in return value.

Reply:
Please change `$tmp` to `${tmp}` to be consistent with the full pattern I've used elsewhere.